addons/web/static/src/core/notifications/notification.xml

```diff
@@ -2,17 +2,15 @@
 <templates xml:space="preserve">
 
     <t t-name="web.NotificationWowl">
-        <div t-on-mouseenter="this.props.freeze" t-on-mouseleave="this.props.refresh" t-attf-class="o_notification {{props.className}} border border-{{props.type}} bg-view mb-2 position-relative"
+        <div t-on-mouseenter="this.props.freeze" t-on-mouseleave="this.props.refresh" t-attf-class="o_notification {{props.className}} bg-view border border-{{props.type}} mb-2 p-3 position-relative rounded shadow-lg"
          role="alert" aria-live="assertive" aria-atomic="true">
-            <strong t-if="props.title" t-attf-class="o_notification_title d-block text-{{props.type}} py-2 ps-3 pe-5" t-out="props.title"/>
-            <button type="button" class="o_notification_close btn" aria-label="Close" t-on-click="props.close">
-                <i class="oi oi-close"/>
-            </button>
-            <div t-attf-class="o_notification_body ps-3 pe-5 py-2">
+            <h5 t-if="props.title" class="o_notification_title d-block m-0 pe-5" t-out="props.title"/>
+            <button type="button" class="o_notification_close btn-close" aria-label="Close" t-on-click="props.close"/>
+            <div t-attf-class="o_notification_body {{props.title ? 'mt-2' : ''}} pe-5">
                 <div t-if="props.message" class="me-auto o_notification_content" t-out="props.message"/>
-                <div t-if="props.buttons.length" class="o_notification_buttons d-flex gap-2 mt-2">
+                <div t-if="props.buttons.length" class="o_notification_buttons d-flex gap-2 mt-3">
                     <button t-foreach="props.buttons" t-as="button" type="button" t-key="button_index"
-                    t-attf-class="btn {{button.primary ? 'btn-primary' : 'btn-secondary'}}" t-on-click="button.onClick">
+                    t-attf-class="btn {{button.primary ? 'btn-' + props.type : 'btn-secondary'}}" t-on-click="button.onClick">
```

Comment:
Imo the "Cancel/Abort" should always be the secondary option: 

If the user wants to perform what we see as a really critical action (e.g.: destroying something) we have to assume it's something he is trying to do and that he has a reason to do it, because the application is allowing him to do it. So, the warning "Are you sure you want to destroy" -> main action should be "Yes, destroy"

Reply:
I'm sorry, the example provided is debatable at least.

Historically we "guide" the user to the safest option by using primary action... but it doesn't fit every scenario and having a danger `["Yes, destroy"]` button would make much more sense indeed, you're right 👍

I think that I was misled by the change in paradigm you're introducing, basically saying that "`primary` ≠ _primary color_" anymore, but "`primary` = _action we expect the user to perform_". 
This changes of perspective highlight a little paradox: 

---

**You destroyed everything**.       

[Undo]

---

[Undo] cannot be red (primary) of course... we could argue that it should be "secondary", but "_secondary to what?_" since these words don't means colors anymore but actions' hierarchy ? 

To summarize, having danger/warning... buttons is ok to me, but the wording and the implementation is misleading.
Does it make sense?